frontend/src/pages/ApartmentPage.tsx

```diff
@@ -41,15 +42,23 @@ const ApartmentPage = (): ReactElement => {
   const [showSignInError, setShowSignInError] = useState(false);
   const [sortBy, setSortBy] = useState<Fields>('date');
   const toastTime = 3500;
-
+  const [notFound, setNotFound] = useState(false);
+  const [notFoundLandlord, setNotFoundLandlord] = useState(0);
+  const handlePageNotFound = () => {
```

Comment:
So actually, I realized the real reason why it's not working rn is because `handlePageNotFound` actually never gets called if `aptId` is not a valid id. This line needs to be added in `app.ts` so that we return an error if we encounter an invalid `aptId`. At the moment, if `aptId` is invalid, this route just returns a JSON object containing that id so the code is never reaching the `catch` clause. 

Basically, if the `aptId` is invalid, we should not even be looking for the landlord. We can actually get rid of everything related to `setNotFoundLandlord`. 
<img width="621" alt="Screen Shot 2021-12-19 at 12 02 51 PM" src="https://user-images.githubusercontent.com/57203639/146683890-f31a2159-44af-47bc-b85a-0587afa1b95b.png">

Reply:
Ok awesome! I thought that was the case so I came up with a patchwork solution, but I think this solution would work better. I'll change it.